File: asgn3/questiontf.cpp
/** ---------------------------------------------
*
* @file		questiontf.cpp
* @author	Nick McCollum
* @version	1.0
*
* CS-202 - Assignment 3 - Abstract Classes
*		   QuestionTF class implementation
*
* -------------------------------------------- */

#include <iostream>
#include "questiontf.h"

using namespace std;

// ----------------------------------------------
QuestionTF::QuestionTF(string questionText = "", string answerText = "")
	: Question(questionText, answerText)
{
	this->questionText = questionText;
	this->answerText = answerText;
	correct = false;
}

// ----------------------------------------------
void QuestionTF::showQuestion() {
	cout << questionText;
	cout << ". (True or False)";
}

// ----------------------------------------------
bool QuestionTF::checkAnswer(string givenAnswer) {
	string actualAnswer = answerText;

	// Remove whitespace and convert to lower
	removeWhitespace(actualAnswer);
	removeWhitespace(givenAnswer);
	toLowerCase(actualAnswer);
	toLowerCase(givenAnswer);

	if (givenAnswer.size() == 1 && givenAnswer.at(0) == actualAnswer.at(0)) {
		markCorrect();
		return true;
	}
	if (givenAnswer.compare(actualAnswer) == 0) {
		markCorrect();
		return true;
	}
	return false;
}

File: Quiz/QuestionTF.java
/**
 * CS-202 - Assignment 5 - Java Classes
 * True / False question object used by Quiz.
 * Displays TF question and checks the user's answer.
 * 
 * @author Nick McCollum
 * @version 1.0
 *
 */

public class QuestionTF extends Question{

	public QuestionTF(String question, String answer) {
		super(question, answer);
		this.question = question;
		this.answer = answer;
		correct = false;
	}

	// ------------------------------------------------------------------
	/**
	 * Displays the TF question text in the console.
	 */
	public void showQuestion() {
		System.out.print(question + ". (True or False)");
	}
	
	// ------------------------------------------------------------------
	/**
	 * Checks if the answer provided by the user was correct.
	 * @param givenAnswer	The answer entered by the user.
	 * @return boolean	True if the answer was correct, false if not.
	 */
	public boolean checkAnswer(String givenAnswer) {
		givenAnswer = removeWhitespace(givenAnswer.toLowerCase());
		String actualAnswer = removeWhitespace(answer.toLowerCase());
		
		if (givenAnswer.length() == 1 && givenAnswer.charAt(0) == actualAnswer.charAt(0)) {
			markCorrect();
			return true;
		}
		if (givenAnswer.contentEquals(actualAnswer)) {
			markCorrect();
			return true;
		}
		
		return false;
	}
}
